server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -2199,10 +2208,16 @@ private Set<DataSegmentPlus> createNewIdsOfAppendSegmentsAfterReplace(
           newInterval = replaceInterval;
           break;
         } else if (replaceInterval.overlaps(oldInterval)) {
-          throw new ISE(
-              "Incompatible segment intervals for commit: [%s] and [%s].",
-              oldInterval, replaceInterval
-          );
+          final String conflictingSegmentId = oldSegment.getId().toString();
+          final String upgradeVersion = upgradeSegmentToLockVersion.get(conflictingSegmentId);
+          throw DruidException.forPersona(DruidException.Persona.OPERATOR)
```

Comment:
curious why this is targeting the cluster `OPERATOR` persona instead of an `ADMIN` or `USER` persona. Since compaction is increasingly becoming an end-user thing entity via compaction supervisor, etc, would it be better to target a non-operator persona?

Reply:
This was chosen because ingestion is associated with an OPERATOR in my opinion.